plugins/rikki/loungestatistics/classes/statistics/Statistics.php

```diff
@@ -0,0 +1,166 @@
+                    $winner = ($game->winner_id == $team->id);
+                    $tO = ($game->team_one_id == $team->id);
+                    
+                    $game->getTeamOneBans()->each( function ($item) use ($tO, &$heroesArray) {
+                        if ($tO) {
+                            $heroesArray[$item->title]['bans_by_team']++;
+                        } else {
+                            $heroesArray[$item->title]['bans_against_team']++;
+                        }
+                    });
+                    $game->getTeamTwoBans()->each( function ($item) use ($tO, &$heroesArray) {
+                        if ($tO) {
+                            $heroesArray[$item->title]['bans_against_team']++;
+                        } else {
+                            $heroesArray[$item->title]['bans_by_team']++;
+                        }
+                    });
+    
+                    $heroesArray[$gP->hero->title]['picks']++;
+                    if ($winner) {
+                        $heroesArray[$gP->hero->title]['wins']++;
+                    }
+
+                    if ($type == "slothAll") {
+                        $hero = $gP->hero;
+                        $heroesArray[$hero->title]['kills'] += $gP->kills;
+                        $heroesArray[$hero->title]['assists'] += $gP->assists;
+                        $heroesArray[$hero->title]['deaths'] += $gP->deaths;
+                        $heroesArray[$hero->title]['siege_dmg'] += $gP->siege_damage;
+                        $heroesArray[$hero->title]['hero_dmg'] += $gP->hero_damage;
+                        $heroesArray[$hero->title]['dmg_taken'] += $gP->damage_taken;
+                        $heroesArray[$hero->title]['healing'] += $gP->healing;
+                        $heroesArray[$hero->title]['xp'] += $gP->experience_contribution;
+                    }
+                }
+            }
+        } else if ($type == "division") {
+            foreach($rawData as $match) {
+                foreach ($match->games as $game) {
+                    $game->getTeamOneBans()->each( function ($item) use (&$heroesArray) {
+                        $heroesArray[$item->title]['bans']++;
+                    });
+                    $game->getTeamTwoBans()->each( function ($item) use (&$heroesArray) {
+                        $heroesArray[$item->title]['bans']++;
+                    });
+                    foreach ($game->gameParticipations as $gP) {
+                        if ($gP->hero == null) {
+                            continue;
+                        }
+
+                        $heroesArray[$gP->hero->title]['picks']++;
+
+                        if ($game->winner_id == $gP->team->id) {
+                            $heroesArray[$gP->hero->title]['wins']++;
+                        }                    
+                    }
+                }
+            }
+        }
+
+        return new Collection($heroesArray);
+    }
```

Comment:
This function is too long overall - but both the Division part and the Sloth part can be split off into their own functions